ec/ecresource/deploymentresource/read.go

```diff
@@ -48,37 +84,86 @@ func readResource(_ context.Context, d *schema.ResourceData, meta interface{}) d
 	})
 	if err != nil {
 		if deploymentNotFound(err) {
-			d.SetId("")
-			return nil
+			diags.AddError("Deployment not found", err.Error())
+			return nil, diags
 		}
-		return diag.FromErr(multierror.NewPrefixed("failed reading deployment", err))
+		diags.AddError("Deloyment get error", err.Error())
+		return nil, diags
 	}
 
-	if !hasRunningResources(res) {
-		d.SetId("")
-		return nil
+	if !utils.HasRunningResources(response) {
+		return nil, nil
+	}
+
+	if response.Resources == nil || len(response.Resources.Elasticsearch) == 0 {
+		diags.AddError("Get resource error", "cannot find Elasticsearch in response resources")
+		return nil, diags
+	}
+
+	if response.Resources.Elasticsearch[0].Info.PlanInfo.Current != nil && response.Resources.Elasticsearch[0].Info.PlanInfo.Current.Plan != nil {
+		if err := checkVersion(response.Resources.Elasticsearch[0].Info.PlanInfo.Current.Plan.Elasticsearch.Version); err != nil {
+			diags.AddError("Get resource error", err.Error())
+			return nil, diags
+		}
+	}
+
+	refId := ""
+
+	var elasticsearchPlan *elasticsearchv2.ElasticsearchTF
+
+	if diags = tfsdk.ValueAs(ctx, plan.Elasticsearch, &elasticsearchPlan); diags.HasError() {
```

Comment:
Why are we reading values from the state in this case. Shouldn't we be reading the API response here?

Reply:
In this line we're reading the current plan. The API response is read in line 134.

Reply:
@tobio you're correct in a sense that it could be state in some cases but otherwise my original answer still states true.

Reply:
All good, I'd confused myself.